src/runtime/runtime.ts

```diff
@@ -36,6 +42,15 @@ export type RuntimeArcOptions = Readonly<{
   inspectorFactory?: ArcInspectorFactory;
 }>;
 
+type SpawnArgs = {
+  id: string,
+  serialization?: string,
+  context: Manifest,
+  composer: SlotComposer,
+  storage: string,
+  portFactories: []
```

Comment:
Can you add a type for this? At the moment this is just `any[]`

Reply:
Typing `pecFactory` was non-trivial, there are conflicting types used on web-platform and node-platform that have the same name. 

I added types where it wasn't too hard, but for a couple of cases I decided the work was out of scope for this PR.